src/main/java/it/gov/pagopa/payhub/activities/activity/treasury/TreasuryOpiIngestionActivityImpl.java

```diff
@@ -26,50 +28,87 @@
-  private final IngestionFlowFileDao ingestionFlowFileDao;
-  private final IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService;
+    private final IngestionFlowFileType ingestionflowFileType;
+    private final IngestionFlowFileDao ingestionFlowFileDao;
+    private final IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService;
+    private final TreasuryUnmarshallerService treasuryUnmarshallerService;
+
+
+    public TreasuryOpiIngestionActivityImpl(
+                                            IngestionFlowFileDao ingestionFlowFileDao,
+                                            IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService,
+                                            TreasuryUnmarshallerService treasuryUnmarshallerService) {
+        this.ingestionflowFileType = IngestionFlowFileType.OPI;
+        this.ingestionFlowFileDao = ingestionFlowFileDao;
+        this.ingestionFlowFileRetrieverService = ingestionFlowFileRetrieverService;
+        this.treasuryUnmarshallerService = treasuryUnmarshallerService;
+    }
 
 
-  public TreasuryOpiIngestionActivityImpl(IngestionFlowFileDao ingestionFlowFileDao,
-                                          IngestionFlowFileRetrieverService ingestionFlowFileRetrieverService) {
-    this.ingestionflowFileType = IngestionFlowFileType.OPI;
-    this.ingestionFlowFileDao = ingestionFlowFileDao;
-    this.ingestionFlowFileRetrieverService = ingestionFlowFileRetrieverService;
-  }
+    @Override
+    public TreasuryIufResult processFile(Long ingestionFlowFileId) {
+        List<String> iufIuvList = new ArrayList<>();
+        List<Path> ingestionFlowFiles = null;
+        AtomicBoolean success = new AtomicBoolean(true);
+        try {
+            IngestionFlowFileDTO ingestionFlowFileDTO = findIngestionFlowFileRecord(ingestionFlowFileId);
 
+            ingestionFlowFiles = retrieveFiles(ingestionFlowFileDTO);
 
-  @Override
-  public TreasuryIufResult processFile(Long ingestionFlowFileId) {
-    List<String> iufIuvList = new ArrayList<>();
-    try {
-      IngestionFlowFileDTO ingestionFlowFileDTO = findIngestionFlowFileRecord(ingestionFlowFileId);
+        } catch (Exception e) {
+            log.error("Error during TreasuryOpiIngestionActivity ingestionFlowFileId {}", ingestionFlowFileId, e);
+            return new TreasuryIufResult(Collections.emptyList(), false);
+        }
 
-      List<Path> ingestionFlowFiles = retrieveFiles(ingestionFlowFileDTO);
+        if (ingestionFlowFiles != null && !ingestionFlowFiles.isEmpty()) {
+            ingestionFlowFiles.forEach(path -> {
+                File ingestionFlowFile = path.toFile();
+                log.debug("file from zip archive with name {} loaded successfully ", ingestionFlowFile.getName());
 
-     log.debug("Successfully retrieved the following files related to the ingestionFlowFileId {}: {}", ingestionFlowFileId, ingestionFlowFiles);
+                success.set(parseData(ingestionFlowFile));
```

Comment:
you should collect iuf list, refactor this stream pipeline to map each file processing in order to build the resulting list (parseData should not return boolean, but the list of iuf.
You should do something like:

ingestionFlowFiles.stream()
   .map(this::parseFile) // log and File conversion, will be performed inside this method
   .flatMap(List::stream)
   .toList()

Reply:
refactor done